src/model/treewalker.js

```diff
@@ -204,33 +204,37 @@ export default class TreeWalker {
 	 */
 	_next() {
 		const previousPosition = this.position;
```

Comment:
Maybe we can drop `position` variable in this function? It is confusing and I am not really sure it is needed.